library/sha512.c

```diff
@@ -150,6 +150,9 @@ int mbedtls_sha512_starts_ret( mbedtls_sha512_context *ctx, int is384 )
     }
     else
```

Comment:
Missing above: `SHA512_VALIDATE_RET` should not accept `is384 == 1` if SHA384 is disabled.